modules/mwOpenLinkIdSystem.js

```diff
@@ -0,0 +1,150 @@
+    utils.logError('User ID - mwOlId submodule requires accountId to be defined');
+    return false;
+  }
+  if (!configParams.partnerId) {
+    utils.logError('User ID - mwOlId submodule requires partnerId to be defined');
+    return false;
+  }
+  return true;
+}
+
+function deserializeMWOlId(mwOLIdStr) {
+  const mwOLId = {};
+  const values = mwOLIdStr.split(',');
+
+  values.forEach(function(value) {
+    const pair = value.split(':');
+    // unpack a value of 1 as true
+    mwOLId[pair[0]] = +pair[1] === 1 ? true : pair[1];
+  });
+
+  return mwOLId;
+}
+
+function serializeMWOLId(mwOLId) {
+  let components = [];
+
+  if (mwOLId.eid) {
+    components.push('eid:' + mwOLId.eid);
+  }
+  if (mwOLId.ibaOptout) {
+    components.push('ibaOptout:1');
+  }
+  if (mwOLId.ccpaOptout) {
+    components.push('ccpaOptout:1');
+  }
+
+  return components.join(',');
+}
+
+function readCookie(name) {
+  if (!name) name = openLinkID.name;
+  const mwOlIdStr = storage.getCookie(name);
+  if (mwOlIdStr) {
+    return deserializeMWOlId(decodeURIComponent(mwOlIdStr));
+  }
+  return null;
+}
+
+function writeCookie(mwOLId) {
+  if (mwOLId) {
+    const mwOLIdStr = encodeURIComponent(serializeMWOLId(mwOLId));
+    storage.setCookie(openLinkID.name, mwOLIdStr, getExpirationDate(), 'lax');
+  }
+}
+
+function register(configParams, olid) {
+  const { accountId, partnerId, uid } = configParams;
+  const url = 'https://ol.mediawallahscript.com/?account_id=' + accountId +
+            '&partner_id=' + partnerId +
+            '&uid=' + uid +
+            '&olid=' + olid +
+            '&cb=' + Math.random()
+            ;
+  ajax(url);
+}
+
+function setID(configParams) {
+  if (!isValidConfig(configParams)) return undefined;
+
+  const mwOLId = readCookie();
+  openLinkID.value = utils.generateUUID();
+
+  const newmwOLId = mwOLId ? utils.deepClone(mwOLId) : {eid: openLinkID.value};
+  writeCookie(newmwOLId);
+  register(configParams, newmwOLId.eid);
+  return {
+    id: mwOLId
+  };
+};
+
+/* End MW */
+
+export { writeCookie };
+
+/** @type {Submodule} */
+export const mwOpenLinkSubModule = {
+  /**
+     * used to link submodule with config
+     * @type {string}
+     */
+  name: 'mwOpenLinkId',
+  /**
+     * decode the stored id value for passing to bid requests
+     * @function
+     * @param {MwOlId} mwOlId
+     * @return {(Object|undefined}
+     */
+  decode(mwOlId) {
+    if (mwOlId && utils.isPlainObject(mwOlId)) {
+      return { mwOlId };
+    }
+    return undefined;
+  },
+
+  /**
+     * performs action to obtain id and return a value in the callback's response argument
+     * @function
+     * @param {SubmoduleParams} [submoduleParams]
+     * @returns {id:MwOlId | undefined}
+     */
+  getId(submoduleConfig) {
+    const submoduleConfigParams = (submoduleConfig && submoduleConfig.params) || {};
+    if (!isValidConfig(submoduleConfigParams)) return undefined;
+    const newId = setID(submoduleConfigParams);
+    return newId;
```

Comment:
is there a reason to set this as a variable vs just returning?

```suggestion
    return setID(submoduleConfigParams);
```